src/test/java/games/strategy/util/VersionTest.java

```diff
@@ -1,106 +1,61 @@
-    final Version v2 = new Version(1, 1, 0);
-    assertNotEquals(v1, v2);
-    assertNotEquals(v2, v1);
-  }
+  public void testCompareTo() {
+    assertThrows(NullPointerException.class, () -> new Version(1, 0).compareTo(null));
 
-  @Test
-  public void testCompare3() {
-    final Version v1 = new Version(0, 0, 0);
-    final Version v2 = new Version(0, 1, 0);
-    assertNotEquals(v1, v2);
-    assertNotEquals(v2, v1);
-  }
+    assertThat(new Version(1, 0, 0, 0).compareTo(new Version(2, 0, 0, 0)), is(lessThan(0)));
+    assertThat(new Version(0, 1, 0, 0).compareTo(new Version(0, 2, 0, 0)), is(lessThan(0)));
+    assertThat(new Version(0, 0, 1, 0).compareTo(new Version(0, 0, 2, 0)), is(lessThan(0)));
+    assertThat(new Version(0, 0, 0, 1).compareTo(new Version(0, 0, 0, 2)), is(lessThan(0)));
+    assertThat(new Version(0, 0, 0, 0).compareTo(new Version("0.0.0.dev")), is(lessThan(0)));
 
-  @Test
-  public void testCompare4() {
-    final Version v1 = new Version(0, 0, 0);
-    final Version v2 = new Version(0, 0, 1);
-    assertNotEquals(v1, v2);
-    assertNotEquals(v2, v1);
-  }
+    assertThat(new Version(1, 0, 0, 0).compareTo(new Version(1, 0, 0, 0)), is(0));
+    assertThat(new Version(0, 1, 0, 0).compareTo(new Version(0, 1, 0, 0)), is(0));
+    assertThat(new Version(0, 0, 1, 0).compareTo(new Version(0, 0, 1, 0)), is(0));
+    assertThat(new Version(0, 0, 0, 1).compareTo(new Version(0, 0, 0, 1)), is(0));
+    assertThat(new Version("0.0.0.dev").compareTo(new Version("0.0.0.dev")), is(0));
 
-  @Test
-  public void testRead1() {
-    assertEquals("1.2.3", new Version("1.2.3").toString());
+    assertThat(new Version(2, 0, 0, 0).compareTo(new Version(1, 0, 0, 0)), is(greaterThan(0)));
+    assertThat(new Version(0, 2, 0, 0).compareTo(new Version(0, 1, 0, 0)), is(greaterThan(0)));
+    assertThat(new Version(0, 0, 2, 0).compareTo(new Version(0, 0, 1, 0)), is(greaterThan(0)));
+    assertThat(new Version(0, 0, 0, 2).compareTo(new Version(0, 0, 0, 1)), is(greaterThan(0)));
+    assertThat(new Version("0.0.0.dev").compareTo(new Version(0, 0, 0, 0)), is(greaterThan(0)));
   }
 
   @Test
-  public void testRead2() {
+  public void testToString() {
+    assertEquals("1.2.3", new Version("1.2.3").toString());
     assertEquals("1.2", new Version("1.2").toString());
-  }
-
-  @Test
-  public void testRead3() {
     assertEquals("1.2", new Version("1.2.0").toString());
-  }
-
-  @Test
-  public void verifyDevVersionToString() {
     assertEquals("1.2.3.dev", new Version("1.2.3.dev").toString());
-    assertEquals("1_2_3_dev", new Version("1.2.3.dev").toStringFull('_'));
   }
 
   @Test
-  public void nullTest() {
-    assertFalse(new Version(1, 2, 3).isCompatible(null));
-    assertFalse(new Version(1, 2, 3).equals(null));
-    assertFalse(new Version(1, 2, 3).isGreaterThan(null));
-    assertFalse(new Version(1, 2, 3).isLessThan(null));
+  public void testToStringFull() {
+    assertEquals("1_2_3_dev", new Version("1.2.3.dev").toStringFull('_'));
```

Comment:
Any reason to not check the other `new Version` constructor variants as well in this test case? My concern is we may not be testing `toStringFull` as thoroughly as we could be otherwise.

Reply:
No reason other than that this was an existing test and had nothing to do with this PR. :smile:  It only appears as a change because it got moved around when I reorganized `VersionTest`.